Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -512,6 +513,31 @@ private extension ProductStore {
         }
         storageProduct.addToTags(NSOrderedSet(array: storageTags))
     }
+
+    /// Updates, inserts, or prunes the provided StorageProduct's downloadable files using the provided read-only Product's downloadable files
+    ///
+    func handleProductDownloadableFiles(_ readOnlyProduct: Networking.Product, _ storageProduct: Storage.Product, _ storage: StorageType) {
+
+        // Removes all the downloadable files first.
+        storageProduct.downloadableFilesArray.forEach { existingStorageDownloadableFile in
+            storage.deleteObject(existingStorageDownloadableFile)
+            storageProduct.removeFromDownloads(existingStorageDownloadableFile)
+        }
+
+        var storageDownloads = [StorageProductDownload]()
```

Comment:
I think this could be an `ordered set` and avoid the conversion in line 539!

Reply:
In the end, we need an `NSOrderedSet` which is not mutable. The only way we can create it is from an Array or `NSMutableOrderedSet`
So we need atleast one conversion. Either `Array` >> `NSOrderedSet` or `NSMutableOrderedSet` >> `Array` >> `NSOrderedSet`

Reply:
I think you can return a `NSMutableOrderedSet` since it inherits from an `NSOrderedSet`. But I guess it's not a big deal.

Reply:
Any change on this @partho-maple?

Reply:
Yeap. pushed.  😛